File: src/main/java/org/kite9/diagram/common/elements/ConstructionElement.java
package org.kite9.diagram.common.elements;

import org.kite9.diagram.model.DiagramElement;

/**
 * Represent all or part of one or more underlying (real) diagram elements.
 * 
 * @author robmoffat
 *
 */
public interface ConstructionElement {
		
	public boolean isPartOf(DiagramElement de);
	
}

File: src/main/kotlin/org/kite9/diagram/common/elements/ConstructionElement.kt
package org.kite9.diagram.common.elements

import org.kite9.diagram.model.DiagramElement

/**
 * Represent all or part of one or more underlying (real) diagram elements.
 *
 * @author robmoffat
 */
interface ConstructionElement {

    fun isPartOf(de: DiagramElement?): Boolean

}